src/profileflow.cpp

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "leakdetector.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "profileflow.h"
+#include "subscriptiondata.h"
+#include "tasks/getsubscriptiondetails/taskgetsubscriptiondetails.h"
+#include "taskscheduler.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+
+namespace {
+Logger logger(LOG_MODEL, "ProfileFlow");
+}
+
+ProfileFlow::ProfileFlow() { MVPN_COUNT_CTOR(ProfileFlow); }
+
+ProfileFlow::~ProfileFlow() { MVPN_COUNT_DTOR(ProfileFlow); }
+
+void ProfileFlow::setState(State state) {
+  logger.debug() << "Set state" << state;
+
+  m_state = state;
+  emit stateChanged(m_state);
+}
+
+void ProfileFlow::start() {
+  logger.debug() << "Start profile flow";
+  reset();
+  setState(StateLoading);
```

Comment:
you can probably remove `reset()` entirely. It does nothing else than setting a state, but then you overwrite it again here.